src/wallet/rpc/spend.cpp

```diff
@@ -1457,6 +1457,102 @@ RPCHelpMan sendall()
     };
 }
 
+RPCHelpMan submitrawtransactiontowallet()
+{
+    return RPCHelpMan{"submitrawtransactiontowallet",
+        "EXPERIMENTAL warning: this call may be changed in future releases.\n"
+        "\nSubmit a raw transaction to the wallet directly, possibly providing initial metadata.\n",
+        {
+            {"hexstring", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The hex string of the raw transaction"},
+            {"options", RPCArg::Type::OBJ_USER_KEYS, RPCArg::Optional::OMITTED_NAMED_ARG, "Options",
+                {
+                    {"foreign_outputs", RPCArg::Type::ARR, RPCArg::Default{UniValue::VARR}, "List of outputs sent by other wallets (eg, for a CoinJoin transaction)",
+                        {
+                            {"output_index", RPCArg::Type::NUM, RPCArg::Optional::OMITTED, "Index of an output to mark as foreign"},
+                        },
+                    },
+                },
+            },
+        },
+        RPCResult{RPCResult::Type::NONE, "", ""},
+        RPCExamples{
+            "\nCreate a transaction\n"
+            + HelpExampleCli("createrawtransaction", "\"[{\\\"txid\\\" : \\\"mytxid\\\",\\\"vout\\\":0}]\" \"{\\\"myaddress\\\":0.01}\"") +
+            "Sign the transaction, and get back the hex\n"
+            + HelpExampleCli("signrawtransactionwithwallet", "\"myhex\"") +
+            "\nSend the transaction (signed hex)\n"
+            + HelpExampleCli("submitrawtransactiontowallet", "\"signedhex\"") +
+            "\nAs a JSON-RPC call\n"
+            + HelpExampleRpc("submitrawtransactiontowallet", "\"signedhex\"")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+        {
+            std::shared_ptr<CWallet> const pwallet{GetWalletForJSONRPCRequest(request)};
+            if (!pwallet) return UniValue::VNULL;
+
+            RPCTypeCheck(request.params, {
+                UniValue::VSTR, // hexstring
+                UniValue::VOBJ, // options
+            }, /*fAllowNull=*/true);
+
```

Comment:
Would this PR be of interest to you?
https://github.com/bitcoin/bitcoin/pull/26039